geode-core/src/test/java/org/apache/geode/internal/statistics/LinuxSystemStatsTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.internal.statistics;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.geode.CancelCriterion;
+import org.apache.geode.Statistics;
+import org.apache.geode.internal.statistics.platform.LinuxProcFsStatistics;
+import org.apache.geode.internal.statistics.platform.LinuxSystemStats;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.tools.ant.filters.StringInputStream;
+import org.junit.*;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+
+/**
+ * Technically a linux only test - the file handling is all mocked up so the test can run on any
+ * host os.
+ */
+@Category(IntegrationTest.class)
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("*.IntegrationTest")
+@PrepareForTest(LinuxProcFsStatistics.class)
+@Ignore
+public class LinuxSystemStatsTest extends StatSamplerTestCase {
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private int[] ints;
+  private long[] longs;
+  private double[] doubles;
+  private LocalStatisticsFactory statisticsFactory;
+  private File testDir;
+
+  @Before
+  public void setUp() throws Exception {
+    this.testDir = this.temporaryFolder.getRoot();
```

Comment:
You can insert one line here to prevent the tests from running on Windows or Solaris (I suspect it fail on those OSes):
```java
    Assume.assumeTrue(SystemUtils.isLinux() || SystemUtils.isMacOSX());
```
The Assume is imported like this:
```java
import org.junit.Assume;
```
Or use it as a static import and remove "Assume." from the line in setUp:
```java
import static org.junit.Assume.assumeTrue;
```

Reply:
The test mocks the linux "/proc/stat" file where the metrics are taken from.   So these test will run on any system with results.   Tested on mac and windows and gave the same deterministic results.

Reply:
Perfect! I forgot you used PowerMock.